compiler/transforms/dispatch_elementwise_mult.py

```diff
@@ -35,67 +38,46 @@ def match_and_rewrite(self, op: linalg_extension.Mul, rewriter: PatternRewriter)
             if not isinstance(inp.type.get_element_type(), builtin.IntegerType):
                 return
 
-        op.library_call = builtin.StringAttr("snax_hwpe_mult")
+        if len(op.iterator_types) != 1:
+            return
 
-        return
+        if op.iterator_types.data[0].data is not linalg.IteratorType.PARALLEL:
+            return
 
+        ## Check if operation is muli
+        ## two operations: first operation is arith.muli, last operation is yield
 
-class LowerLinalgToFunc(RewritePattern):
-    """
-    Lowers linalg.mul functions marked with a library call to function calls
-    """
+        mult_op = op.body.block.first_op
+        yield_op = op.body.block.last_op
 
-    @op_type_rewrite_pattern
-    def match_and_rewrite(self, op: linalg_extension.Mul, rewriter: PatternRewriter):
-        if op.library_call is None:
+        # last operation is linalg.yield
+        if not isinstance(yield_op, linalg.YieldOp):
+            return
```

Comment:
I don't think it's required to test this.
A linalg generic's body is required to have a yield operation, because otherwise it should trip the verifier (has no terminator)

Reply:
Oh I think I see why, you need this for your later test I guess?